diesel/src/mysql/types/mod.rs

```diff
@@ -5,39 +5,231 @@ mod date_and_time;
+        let mut bytes = not_none!(bytes);
+        debug_assert!(
+            bytes.len() <= 8,
+            "Received more than 8 bytes decoding u64. \
+             Was the expression accidentally identified as BigInt?"
+        );
+        debug_assert!(
+            bytes.len() >= 8,
+            "Received fewer than 8 bytes decoding u64. \
+             Was a Integer expression accidentally identified as BigInt?"
+        );
+        bytes
+            .read_u64::<<Mysql as Backend>::ByteOrder>()
+            .map_err(Into::into)
+    }
+}
+
+impl ToSql<Bool, Mysql> for bool {
     fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
         let int_value = if *self { 1 } else { 0 };
-        <i32 as ToSql<sql_types::Integer, Mysql>>::to_sql(&int_value, out)
+        <i32 as ToSql<Integer, Mysql>>::to_sql(&int_value, out)
     }
 }
 
-impl FromSql<sql_types::Bool, Mysql> for bool {
+impl FromSql<Bool, Mysql> for bool {
     fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
         Ok(not_none!(bytes).iter().any(|x| *x != 0))
     }
 }
 
+impl<ST> HasSqlType<Unsigned<ST>> for Mysql
+where
+    ST: NotNull + SingleValue + QueryId,
+    Mysql: HasSqlType<ST>,
+{
+    fn metadata(lookup: &()) -> MysqlType {
+        <Mysql as HasSqlType<ST>>::metadata(lookup)
+    }
+}
+
+impl<ST> QueryId for Unsigned<ST>
+where
+    ST: NotNull + SingleValue + QueryId,
+    <ST as QueryId>::QueryId: SingleValue + NotNull + QueryId,
+{
+    type QueryId = Unsigned<<ST as QueryId>::QueryId>;
+}
+
+impl QueryFragment<Mysql> for u16 {
+    fn walk_ast(&self, mut out: AstPass<Mysql>) -> QueryResult<()> {
+        let value = *self as i16;
+        out.push_sql(&value.to_string());
+        Ok(())
+    }
+}
+
+impl QueryFragment<Mysql> for u32 {
+    fn walk_ast(&self, mut out: AstPass<Mysql>) -> QueryResult<()> {
+        let value = *self as i32;
+        out.push_sql(&value.to_string());
+        Ok(())
+    }
+}
+
+impl QueryFragment<Mysql> for u64 {
+    fn walk_ast(&self, mut out: AstPass<Mysql>) -> QueryResult<()> {
+        let value = *self as i64;
+        out.push_sql(&value.to_string());
+        Ok(())
+    }
+}
+
+impl Expression for u16 {
+    type SqlType = Unsigned<SmallInt>;
+}
+
+impl Expression for u32 {
+    type SqlType = Unsigned<Integer>;
+}
+
+impl Expression for u64 {
+    type SqlType = Unsigned<BigInt>;
+}
+
+impl SelectableExpression<()> for u16 {}
+impl SelectableExpression<()> for u32 {}
+impl SelectableExpression<()> for u64 {}
+
+impl AppearsOnTable<()> for u16 {}
+impl AppearsOnTable<()> for u32 {}
+impl AppearsOnTable<()> for u64 {}
+
+impl QueryId for u16 {
+    type QueryId = ();
+}
+
+impl QueryId for u32 {
+    type QueryId = ();
+}
+
+impl QueryId for u64 {
+    type QueryId = ();
+}
+
+impl Queryable<Unsigned<SmallInt>, Mysql> for u16 {
+    type Row = u16;
+
+    fn build(row: Self::Row) -> Self {
+        row
+    }
+}
```

Comment:
Here it seems like the impl for `u32` is missing?

Reply:
I went to add this and doing so causes a compilation error. It looks like this implementation is already done by https://github.com/diesel-rs/diesel/blob/master/diesel/src/type_impls/primitives.rs#L38

Reply:
I'll add similar structs to `/diesel/src/type_impls/primitives.rs` for `u16` and `u64`. I imagine that's better than implementing the traits manually here.